internal/handlers/pg/pgdb/settings.go

```diff
@@ -42,193 +38,120 @@ const (
-// That table consists of a single document with settings.
-func createSettingsTable(ctx context.Context, tx pgx.Tx, db string) error {
-	tables, err := tables(ctx, tx, db)
-	if err != nil {
-		return lazyerrors.Error(err)
-	}
-
-	if slices.Contains(tables, settingsTableName) {
-		return ErrAlreadyExist
-	}
-
-	// TODO use common code for tables/collections: use _jsonb, do not use explicit `CREATE TABLE` SQL there, etc.
-	sql := fmt.Sprintf(`CREATE TABLE %s (settings jsonb)`, pgx.Identifier{db, settingsTableName}.Sanitize())
-	_, err = tx.Exec(ctx, sql)
-	if err != nil {
-		pgErr, ok := err.(*pgconn.PgError)
-		if !ok {
-			return lazyerrors.Errorf("pgdb.createSettingsTable: %w", err)
-		}
-
-		switch pgErr.Code {
-		case pgerrcode.InvalidSchemaName:
-			return ErrTableNotExist
-		case pgerrcode.DuplicateTable:
-			return ErrAlreadyExist
-		case pgerrcode.UniqueViolation, pgerrcode.DuplicateObject:
-			// https://www.postgresql.org/message-id/CA+TgmoZAdYVtwBfp1FL2sMZbiHCWT4UPrzRLNnX1Nb30Ku3-gg@mail.gmail.com
-			// Reproducible by integration tests.
-			return newTransactionConflictError(err)
-		default:
-			return lazyerrors.Errorf("pgdb.createSettingsTable: %w", err)
-		}
-	}
-
-	settings := must.NotFail(types.NewDocument("collections", must.NotFail(types.NewDocument())))
-	sql = fmt.Sprintf(`INSERT INTO %s (settings) VALUES ($1)`, pgx.Identifier{db, settingsTableName}.Sanitize())
-	_, err = tx.Exec(ctx, sql, must.NotFail(pjson.Marshal(settings)))
-	if err != nil {
-		return lazyerrors.Error(err)
-	}
-
-	return nil
-}
-
-// getTableName returns the name of the table for given collection or error.
-// If the settings table doesn't exist, it will be created.
-// If the record for collection doesn't exist, it will be created.
-func getTableName(ctx context.Context, tx pgx.Tx, db, collection string) (string, error) {
-	schemaExists, err := schemaExists(ctx, tx, db)
-	if err != nil {
-		return "", lazyerrors.Error(err)
-	}
+// ensureSettings returns PostgreSQL table name for the given FerretDB database and collection names.
+// If such settings don't exist, it creates them, including the creation of the PostgreSQL schema if needed.
+// If settings were created, it returns true as the second return value. If settings already existed, it returns false.
+//
+// It makes a document with _id and table fields and stores it in the settingsTableName table.
+// The given FerretDB collection name is stored in the _id field,
+// the corresponding PostgreSQL table name is stored in the table field.
+// For _id field it creates unique index.
+//
+// It returns a possibly wrapped error:
+//   - ErrInvalidDatabaseName - if the given database name doesn't conform to restrictions.
+//   - *transactionConflictError - if a PostgreSQL conflict occurs (the caller could retry the transaction).
+func ensureSettings(ctx context.Context, tx pgx.Tx, db, collection string) (tableName string, created bool, err error) {
```

Comment:
A potential alternative (proposed by Patryk) is to user the word "metadata" instead of "settings". I like that idea.